### PR TITLE
DOC-475: PXC Security Guide initial draft

### PR DESCRIPTION
**Pull request by azhebel**

### CODE REVIEW

File: doc/source/index.rst [hunk 1]
```diff
@@ -10,22 +10,41 @@ and provides linear scalability for a growing environment.
...
+* **Configuration script for ProxySQL**
+
+  Percona provides a ProxySQL package with the ``proxysql-admin`` tool
+  that automatically configures |PXC| nodes.
+
+* **Automatic configuration of SSL encryption**
+
+  |PXC| includes the ``pxc-encrypt-cluster-traffic`` variable
+  that enables automatic configuration of SSL encrytion.
+
```
Comment (kbauskar) [inline doc/source/index.rst:47]: Performance Optimized for your production workload
|PXC| is performance optimized to handle growing production workload. It offers best in class performance (w.r.t to all its competitors based on same technology). More details check here. (link to blog).
....

File: doc/source/security/index.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+* :ref:`encrypt-traffic`
+
+  Unencrypted traffic can potentially be viewed
+  by anyone monitoring your network.
+  You should enable encryption for all nodes in the cluster to prevent this.
+
+* Data-at-rest encryption?
+
+  Should we cover this? It seems MariaDB has it covered but nothing in PS docs.
```
Comment (kbauskar) [inline doc/source/security/index.rst]: We should reframe it saying
"PXC-5.7 added support for encrypted tablespace introduced in MySQL-5.7 there-by allowing user to encrypt data at rest using MySQL keyring plugin/technique".

**Review by kbauskar — APPROVED** [review]
Approved with 2 small comments.

Please take Kenn approval too.

File: doc/source/security/secure-network.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
+.. _secure-network:
+
+====================
+Securing the Network
+====================
+
+By default, anyone with access to your network can connect to any |PXC| node
+either as a client or as another node joining the cluster.
+This could potentially let them query your data or get a comlete copy of it.
```
Comment (kennt-percona) [inline doc/source/security/secure-network.rst]: "comlete" should be "complete"

File: doc/source/security/encrypt-traffic.rst [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+run the following command::
+
+ $ openssl verify -CAfile ca.pem server-cert.pem client-cert.pem
+
+If the verification is successful, you should see the following output::
+
+ server-cert.pem: OK
+ client-cert.pem: OK
+
+Diploying Keys and Certificates
```
Comment (kennt-percona) [inline doc/source/security/encrypt-traffic.rst]: "Diploying" should be "Deploying"

File: doc/source/security/encrypt-traffic.rst [hunk 5]
```diff
@@ -0,0 +1,446 @@
...
+-------------------------------
+
+Use a secure method (for example, ``scp`` or ``sftp``)
+to send the key and certificate files to each node.
+Place them under the :file:`/etc/mysql/certs/` directory
```
Comment (kennt-percona) [inline doc/source/security/encrypt-traffic.rst:114]: We should mention the permissions that should be on this directory, most likely read-only by mysql, no other access, so probably x400

File: doc/source/index.rst (hunk 1 above)
Comment (azhebel) [reply doc/source/index.rst:47]: @kbauskar which one?
https://www.percona.com/blog/2017/04/19/performance-improvements-percona-xtradb-cluster-5-7-17/
https://www.percona.com/blog/2017/04/19/how-we-made-percona-xtradb-cluster-scale/
Link both posts?

File: doc/source/security/index.rst (hunk 2 above)
Comment (azhebel) [reply doc/source/security/index.rst]: @kbauskar I meant maybe I should add a DOC issue for adding this to PS? Then we can have a link to it from here.

File: doc/source/security/encrypt-traffic.rst (hunk 5 above)
Comment (azhebel) [reply doc/source/security/encrypt-traffic.rst:114]: @kennt-percona could you elaborate? Only read permissions for the mysql user and no one else?

**Comment by kennt-percona** [comment]
I remember seeing some mention of this, but couldn't find it.  What I did find was this:

https://dev.mysql.com/doc/refman/5.7/en/creating-rsa-files-using-openssl.html

This recommends 400 for the private key. It also recommends 444 for the public key.

For our use, I think having all the files be 0400 (read-only by mysql) is ok.  Note that the directory and all the files should be owned by mysql:mysql.

File: doc/source/index.rst (hunk 1 above)
Comment (kbauskar) [reply doc/source/index.rst:47]: first one is fine

**Review by kbauskar — APPROVED** [review]

**Pull request merged**